### DISCUSSION TITLE
Submitting workflow from template via custom event

### DISCUSSION THREAD

**Discussion by nick-kanakis** [argoproj/argo-workflows#4747] [Q&A] [answered]
Hi, I'm trying the following flow.
A service that runs within the same namespace as argo-server reacts to some external events and submits an http request to argo-server to start a new workflow from a workflow-template, based on the [event api](https://argoproj.github.io/argo/events/#events). While I follow the guide I have the following error:
`no valid authentication methods found for mode client`
If I try the same request without the Auth header I get an error to create the `argo-workflows-webhook-clients` secret but this is for github/gitlab... webhooks that is not my case.

Is there an example or guide for how I can submit an http request and create a workflow based on a worklow-template?

**Comment by alexec** [comment, marked as answer]
You are getting the "no valid auth modes" error because you have not started argo-server with the `--auth-mode=client` flag. I don't understand the other error - the argo-server does not need permission to create secrets, only get/list them.

**Reply by jrajasekaran** [reply]
I did not change auth-mode=client, just using the authorization token worked, even with auth-mode=server. Saved my day! thanks
